[nnc] Use given kernel function name while emitting code

navahgar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #66218
* #66217
* __->__ #66216

The `Codegen` object has a `kernel_func_name_` member which should be used as the name for the wrapper function we create during `LLVMCodeGen`.

Differential Revision: [D31445799](https://our.internmc.facebook.com/intern/diff/D31445799)

navahgar committed:
[nnc] Use given kernel function name while emitting code

[ghstack-poisoned]

navahgar commented:
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31445799).

ZolotukhinM reviewed (approved):
Looks good!

priyaramani reviewed (approved):
Thanks for doing this!

navahgar committed:
Update on "[nnc] Use given kernel function name while emitting code"


The `Codegen` object has a `kernel_func_name_` member which should be used as the name for the wrapper function we create during `LLVMCodeGen`.

Differential Revision: [D31445799](https://our.internmc.facebook.com/intern/diff/D31445799)

[ghstack-poisoned]
